src/mvs/fusion.cc

```diff
@@ -151,12 +157,27 @@ void StereoFusion::Run() {
   }
 
   workspace_options.max_image_size = options_.max_image_size;
+  // Saving calculated maps only when the option is set (useful for debugging)
+  // or when using cache to avoid potential multiple calculations of the same
+  // map.
+  workspace_options.save_calculated_maps =
+      options_.save_calculated_maps || options_.use_cache;
   workspace_options.image_as_rgb = true;
   workspace_options.cache_size = options_.cache_size;
   workspace_options.workspace_path = workspace_path_;
   workspace_options.workspace_format = workspace_format_;
   workspace_options.input_type = input_type_;
 
+  const auto image_names = ReadTextFileLines(JoinPaths(
+      workspace_path_, workspace_options.stereo_folder, "fusion.cfg"));
+  int num_threads = 1;
+  if (options_.use_cache) {
+    workspace_.reset(new Workspace(workspace_options));
+  } else {
+    workspace_.reset(new NoCacheWorkspace(workspace_options));
+    workspace_->Load(image_names);
+    num_threads = omp_get_max_threads();
```

Comment:
It would be nice to make the number of threads configurable in the options. It could be initialized to -1 and you use `GetEffectiveNumThreads` to determine the number of threads.

Reply:
Yes, that's a good suggestion and an easy change to make :)